MCU tests change to simulating unitary instead of statevector

alfrisch opened the pull request:
this PR addresses issue #540 

### Summary

`test_mcu1.py` and `test_mcu3.py` are now checking the correctness of the unitary instead of the statevector.
The testing scope is reduced from 7 to 6 control qubits max due to increased testing time.

hushaohan commented:
@alfrisch would it make sense to also randomize the angle params in addition to the current fixed special ones?

alfrisch commented:
@hushaohan I added randomized angles like `np.random.random(1)[0] * pi` and discovered that the graycode implementation of `mcu3` can not work for arbitrary rotations, so I introduced `mcrx`, `mcry` (which I merged with existing implementation from `controlled_ry_gates.py`) and `mcrz` see discussion in issue #540
